airbyte_cdk/sources/declarative/incremental/concurrent_partition_cursor.py

```diff
@@ -99,19 +99,29 @@ def __init__(
         self._semaphore_per_partition: OrderedDict[str, threading.Semaphore] = OrderedDict()
 
         # Parent-state tracking: store each partition’s parent state in creation order
-        self._partition_parent_state_map: OrderedDict[str, Mapping[str, Any]] = OrderedDict()
+        self._partition_parent_state_map: OrderedDict[str, tuple[Mapping[str, Any], int]] = (
+            OrderedDict()
+        )
+        self._parent_state: Optional[StreamState] = None
 
+        # Tracks when the last slice for partition is emitted
         self._finished_partitions: set[str] = set()
+        # Used to track the sequence numbers of open partitions
+        self._open_seqs: deque[int] = deque()
```

Comment:
I think it would be worth defining "open" and "sequence". Here is an attempt at it:
* open means that the not all stream slices for a specific partition key have been processed
* sequence is an number that is being incremented every time we generate stream slices based on a partition. In other words, it is the number of parent records being processed

I'm a bit wary of using the word `sequence` not because I don't think it reflects what it is but rather because we had yet another concept to this class which adds cognitive load. If the definition above is correct, should we change the naming of `next_seq` for `parent_record_count` and `open_seqs` to `processing_parent_indices` or something like that?